src/youtube.coffee

```diff
@@ -24,11 +27,21 @@ module.exports = (robot) ->
         key: process.env.HUBOT_YOUTUBE_API_KEY
       })
       .get() (err, res, body) ->
-        videos = JSON.parse(body)
+        robot.logger.debug body
+        if err
+          robot.logger.error err
+          return msg.send err
+        try
+          videos = JSON.parse(body)
+        catch error
+          robot.logger.error error
+          return msg.send "Error! #{body}"
+        if videos.error
+          robot.logger.error videos.error
+          return msg.send "Error! #{JSON.stringify(videos.error)}"
+        robot.logger.debug videos
```

Comment:
Do you need to `inspect` or `JSON.stringify`? I don't think it'd output what you expect otherwise.